### PR TITLE
RDBC-653 Don't consider raw objects dirty

### PR DESCRIPTION
**Pull request by poissoncorp**
https://issues.hibernatingrhinos.com/issue/RDBC-653/Raw-object-literal-in-node.js-is-always-considered-dirty

Raw TypeScript objects appear to have a `Raven-Node-Type` metadata field set to `null`. This caused spurious updates every time. I separated this corner case from the usual `updateMetadataModifications` flow and added a test for that.

### CODE REVIEW

File: test/Issues/RDBC_653.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import {IDocumentStore} from "../../src";
+import { testContext, } from "../Utils/TestUtil";
+import {assertThat} from "../Utils/AssertExtensions";
+
+describe("RDBC-653", function () {
+
+    let store: IDocumentStore;
+
+    beforeEach(async function () {
```
Comment (ml054) [inline test/Issues/RDBC_653.ts:9]: please add more test cases with falsy values in metadata and object.

Comment (poissoncorp) [reply test/Issues/RDBC_653.ts:9]: done

File: test/Issues/RDBC_653.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            await session.saveChanges();
+            assertThat(session.advanced.numberOfRequests).isEqualTo(1);
+        }
+        finally {
+            session.dispose();
+        }
+
+    })
+
+    it("custom metadata wont be updated spuriously on falsy value 1", async () => {
```
Comment (ml054) [inline test/Issues/RDBC_653.ts:83]: can you include happy path case, like: myMetadata 2 or "abc"?

File: src/Documents/Session/InMemoryDocumentSessionOperations.ts [hunk 3]
```diff
@@ -1773,9 +1773,18 @@ export abstract class InMemoryDocumentSessionOperations
                 dirty = true;
             }
 
-            for (const prop of Object.keys(documentInfo.metadataInstance)) {
+            const metadataKeysToCheck = Object.keys(documentInfo.metadataInstance);
+
+            const ravenNodeTypeKey = CONSTANTS.Documents.Metadata.RAVEN_JS_TYPE;
+            if (metadataKeysToCheck.includes(ravenNodeTypeKey) && TypeUtil.isNullOrUndefined(documentInfo.metadataInstance[ravenNodeTypeKey]))
+            {
+                metadataKeysToCheck.splice(metadataKeysToCheck.indexOf("Raven-Node-Type"), 1);
```
Comment (ml054) [inline src/Documents/Session/InMemoryDocumentSessionOperations.ts:1781]: ?

**Review by ml054 — COMMENTED** [review]
@poissoncorp let's have a quick chat about that PR

File: test/Issues/RDBC_653.ts (hunk 2 above)
Comment (poissoncorp) [reply test/Issues/RDBC_653.ts:83]: done

**Comment by ml054** [comment]
Closed in favor of: https://github.com/ravendb/ravendb-nodejs-client/pull/348
